test/test_nn.py

```diff
@@ -1654,8 +1655,16 @@ def assign_weight():
         # This should work though
         l2.weight = Parameter(torch.randn(10, 10))
 
-    def test_clip_grad_norm(self):
-        l = nn.Linear(10, 10)
+    @parametrize_test('device_type', ('cpu', 'cuda'))
```

Comment:
Another nit! We already have perdevice test classes where device will be auto-parametrized (it is provided as an argument to the test function). See `test_affine_2d_rotate90` as an example. You can move these tests under NNDeviceType if they aren't already and we can remove the logic where we have to skip based on device_type.

Reply:
Done in https://github.com/pytorch/pytorch/pull/91846/commits/adde43969b2e0fb04e313660bf6373d7879e6db5 but now the tests don't run when I locally call pytest

Reply:
huh i don't use pytest but weirdddd @clee2000 do you know how to run specific tests in pytest once they're subscribed to a device type class?

Reply:
ok my bad they actually run as expected with `pytest -k clip_grad test/test_nn.py`, not sure what I did wrong before